src/amr/messengers/hybrid_hybrid_messenger_strategy.hpp

```diff
@@ -160,46 +171,52 @@ namespace amr
             std::unique_ptr<HybridMessengerInfo> hybridInfo{
                 dynamic_cast<HybridMessengerInfo*>(fromFinerInfo.release())};
 
+            auto b_id = resourcesManager_->getID(hybridInfo->modelMagnetic);
+
+            if (!b_id)
+            {
+                throw std::runtime_error(
+                    "HybridHybridMessengerStrategy: missing magnetic field variable IDs");
+            }
+
+            magneticRefinePatchStrategy_.registerIDs(*b_id);
 
-            std::shared_ptr<SAMRAI::xfer::VariableFillPattern> xVariableFillPattern
-                = std::make_shared<XVariableFillPattern>();
+            BalgoPatchGhost.registerRefine(*b_id, *b_id, *b_id, BfieldRefineOp_,
+                                           nonOverwriteInteriorTFfillPattern);
 
-            std::shared_ptr<SAMRAI::xfer::VariableFillPattern> yVariableFillPattern
-                = std::make_shared<YVariableFillPattern>();
 
-            std::shared_ptr<SAMRAI::xfer::VariableFillPattern> zVariableFillPattern
-                = std::make_shared<ZVariableFillPattern>();
+            BregridAlgo.registerRefine(*b_id, *b_id, *b_id, BfieldRegridOp_,
+                                       overwriteInteriorTFfillPattern);
 
-            auto bx_id = resourcesManager_->getID(hybridInfo->modelMagnetic.xName);
-            auto by_id = resourcesManager_->getID(hybridInfo->modelMagnetic.yName);
-            auto bz_id = resourcesManager_->getID(hybridInfo->modelMagnetic.zName);
+            auto e_id = resourcesManager_->getID(hybridInfo->modelElectric);
 
-            if (!bx_id or !by_id or !bz_id)
+            if (!e_id)
             {
                 throw std::runtime_error(
-                    "HybridHybridMessengerStrategy: missing magnetic field variable IDs");
+                    "HybridHybridMessengerStrategy: missing electric field variable IDs");
             }
 
-            magneticRefinePatchStrategy_.registerIDs(*bx_id, *by_id, *bz_id);
-
-            Balgo.registerRefine(*bx_id, *bx_id, *bx_id, BfieldRefineOp_, xVariableFillPattern);
-            Balgo.registerRefine(*by_id, *by_id, *by_id, BfieldRefineOp_, yVariableFillPattern);
-            Balgo.registerRefine(*bz_id, *bz_id, *bz_id, BfieldRefineOp_, zVariableFillPattern);
+            EalgoPatchGhost.registerRefine(*e_id, *e_id, *e_id, EfieldRefineOp_,
+                                           nonOverwriteInteriorTFfillPattern);
 
+            auto e_reflux_id = resourcesManager_->getID(hybridInfo->refluxElectric);
```

Comment:
the `resourcesManager_->getIDsList` will dereference the optional or throw if missing

and it really never should be missing